workspace/api/openfdd_bridge/rule_store.py

```
@@ -235,14 +246,25 @@ def normalize_rule(entry: dict[str, Any], *, saved_by: str = "operator") -> dict
     code = str(entry.get("code") or "")
+    sql = str(entry.get("sql") or "").strip()
+    fault_column = str(entry.get("fault_column") or "fault").strip() or "fault"
     path = str(entry.get("source_path") or "")
-    if path:
-        disk = read_source(path)
-        if disk.strip():
-            code = disk
-    if not code.strip():
-        raise ValueError("rule code is required")
-    _lint_rule_code(code, mode=mode)
+    resolved_backend = backend
```

Ruff (0.15.15)

[error] Undefined name `backend`

(F821)